server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RandomStoring.java

```diff
@@ -87,62 +88,51 @@ public String getMailetInfo() {
 
     @Override
     public void init() throws MessagingException {
+        perUserMailboxCount = MailetUtil.getInitParameterAsStrictlyPositiveInteger(getInitParameter(PER_USER_MAILBOX_COUNT), 10);
     }
 
     private Collection<ReroutingInfos> generateRandomMailboxes() {
-        List<ReroutingInfos> reroutingInfos = this.reroutingInfos.block();
+        List<User> users = cachedUsers.block();
 
         // Replaces Collections.shuffle() which has a too poor statistical distribution
         return ThreadLocalRandom
             .current()
-            .ints(0, reroutingInfos.size())
-            .mapToObj(reroutingInfos::get)
+            .ints(0, users.size())
+            .mapToObj(users::get)
             .distinct()
             .limit(randomRecipientsNumbers.get())
+            .map(user -> new ReroutingInfos(chooseRandomMailboxName(), user))
             .collect(Guavate.toImmutableSet());
     }
 
-    private List<ReroutingInfos> retrieveReroutingInfos() throws UsersRepositoryException {
-        return Streams.stream(usersRepository.list())
-            .map(User::fromUsername)
-            .flatMap(this::buildReRoutingInfos)
-            .collect(Guavate.toImmutableList());
+    private String chooseRandomMailboxName() {
```

Comment:
your "share nothing" design is based on the assumption you know exactly the names of mailboxes expected by the tests. It breaks as soon as you don't use these not-so-random mailbox names.